builder/builder_windows.go

```diff
@@ -0,0 +1,320 @@
+		// 		return fmt.Errorf("Failed to create network: %s, err: %v, msg: %s", network.Name, err, msg)
+		// 	}
+		// 	log.Printf("Successfully set up Docker network: %s\n", network.Name)
+		// }
+
+		log.Println("Setting up Docker configuration...")
+		timeout := time.Duration(configTimeoutInSec) * time.Second
+		configCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		if err := b.setupConfig(configCtx); err != nil {
+			return err
+		}
+		log.Printf("Successfully set up Docker configuration")
+		if task.UsingRegistryCreds() {
+			log.Printf("Logging in to registry: %s\n", task.RegistryName)
+			timeout := time.Duration(loginTimeoutInSec) * time.Second
+			loginCtx, cancel := context.WithTimeout(ctx, timeout)
+			defer cancel()
+			if err := b.dockerLoginWithRetries(loginCtx, task.RegistryName, task.RegistryUsername, task.RegistryPassword, 0); err != nil {
+				return err
+			}
+			log.Println("Successfully logged in")
+		}
+	}
+
+	var completedChans []chan bool
+	errorChan := make(chan error)
+	for _, n := range task.Dag.Nodes {
+		completedChans = append(completedChans, n.Value.CompletedChan)
+	}
+
+	for _, n := range task.Dag.Root.Children() {
+		go b.processVertex(ctx, task, task.Dag.Root, n, errorChan)
+	}
+
+	// Block until either:
+	// - The global context expires
+	// - A step has an error
+	// - All steps have been processed
+	for _, ch := range completedChans {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ch:
+			continue
+		case err := <-errorChan:
+			return err
+		}
+	}
+
+	var deps []*image.Dependencies
+	for _, step := range task.Steps {
+		log.Printf("Step id: %v marked as %v (elapsed time in seconds: %f)\n", step.ID, step.StepStatus, step.EndTime.Sub(step.StartTime).Seconds())
+
+		if len(step.ImageDependencies) > 0 {
+			log.Printf("Populating digests for step id: %s...\n", step.ID)
+			timeout := time.Duration(digestsTimeoutInSec) * time.Second
+			digestCtx, cancel := context.WithTimeout(ctx, timeout)
+			defer cancel()
+			if err := b.populateDigests(digestCtx, step.ImageDependencies); err != nil {
+				return err
+			}
+			log.Printf("Successfully populated digests for step id: %s\n", step.ID)
+			deps = append(deps, step.ImageDependencies...)
+		}
+	}
+
+	if len(deps) > 0 {
+		bytes, err := json.Marshal(deps)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal image dependencies")
+		}
+		fmt.Println("The following dependencies were found:")
+		fmt.Println(string(bytes))
+	}
+
+	return nil
+}
+
+// CleanTask iterates through all build steps and removes
+// their corresponding containers.
+func (b *Builder) CleanTask(ctx context.Context, task *graph.Task) {
+	args := []string{"docker", "rm", "-f"}
+	for _, n := range task.Dag.Nodes {
+		step := n.Value
+		if step.StepStatus != graph.Skipped {
+			killArgs := append(args, step.ID)
+			_ = b.procManager.Run(ctx, killArgs, nil, nil, nil, "")
+		}
+	}
+
+	// for _, network := range task.Networks {
```

Comment:
Delete this file; rename the other one back to what it was; go back to the previous one and add `if runtime.GOOS != "windows"` or make a helper file: `network_windows.go` and invoke this there, where `_windows` would just be a `return` no-op.